docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+
+### Transactions submitted by user
+To query for all transactions submitted by specific user, we use `/provenance/data/tx/{user}` GET query.
+
+**Sign json marshaled query**
+```sh
+bin/signer -data '{"user_id":"alice","target_user_id":"alice"}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Signature**
+```
+MEYCIQDNvPHvfb0GpBpbQ5Gedm0LmEQZbijkmf3vNt4JbuSr7gIhAJMp2cVFeckAQbLSMQg7Fn5vCMyTjyS4/Z7740eZalTj
+```
+
+**Submit query**
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEYCIQDNvPHvfb0GpBpbQ5Gedm0LmEQZbijkmf3vNt4JbuSr7gIhAJMp2cVFeckAQbLSMQg7Fn5vCMyTjyS4/Z7740eZalTj" \
+     -X GET http://127.0.0.1:6001/provenance/data/tx/alice | jq .
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "txIDs": [
+      "1b6d6414-9b58-45d0-9723-1f31712add81",
+      "Tx000",
+      "Tx001"
+    ]
+  },
+  "signature": "MEUCIQDoeIihFrgwmOk2cQdOUsnlZ7gbDcmuGehkH0y0AYGo/AIgNk+Cy3mTUrGSytj5RIg4jHia/ceAUQ5TW5Q1RmG0FuQ="
+}
+```
+
+As you can see, user `alice` submitted 3 tx.
+
+### Query for users reads
+
+To query what keys were read by specific user, including keys versions, we use `/provenance/data/read/{user}`GET query.
+
+**Sign json marshalled query**
+```sh
+bin/signer -data '{"user_id":"alice","target_user_id":"alice"}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Signature**
+```
+MEUCIQCzRD+rS8043PgFRfwVFYKkT4FN+2KCPqU8OCciXlTz/QIgSDvMl/q+AiyeoZI3qDs87y4ScJgVW2e/6dXtnlTQaCQ=
+```
+
+**Submit query**
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEUCIQCzRD+rS8043PgFRfwVFYKkT4FN+2KCPqU8OCciXlTz/QIgSDvMl/q+AiyeoZI3qDs87y4ScJgVW2e/6dXtnlTQaCQ=" \
+     -X GET http://127.0.0.1:6001/provenance/data/read/alice | jq .
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "KVs": [
+      {
+        "key": "key1",
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "key": "key2",
+        "value": "dGhpcyBpcyBhIHNlY29uZCB2YWx1ZQ==",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEUCIAPDXNsnoOCc+PisOiAmvk3giXxNfynW9YjAbcGSvPDaAiEAnKYp2K4WonHofJQ1ag3aYITcdfAXIDwNdggjSwkXS1w="
+}
+```
+
+We can see that user `alice` read two keys, that happens during `Tx001`.
```

Comment:
We can see that user `alice` read two keys, during transaction `Tx001`.

(Describe in brief the format of expected output before the example.... true for all queries)